### PR TITLE
fix(Client): user and url flow

### PR DESCRIPTION
**Pull request by Noggling**
# Description
- Added initial code of conduct
- Added initial contribution
- Changes som routing 
- added navigation to context change
- Renamed Context-page to Project Page

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] Documentation and comments is added where needed.
- [ ] My code is covered by tests.
- [ ] UX has reviewed relevant UI contributions.

## Definition of Done
- [x] PR title and description are to the point and understandable
- [x] I have performed a self-review of my own code

### CODE REVIEW

File: client/packages/portal-client/src/components/portal-router/PortalRouter.tsx [hunk 1]
```diff
@@ -20,13 +20,35 @@ const routes: RouteObject[] = [
...
+					{
+						path: ':contextId/*',
+						element: <AppLoader />,
+						errorElement: <PortalMessagePage title="Fail to load application" type={'Error'} />,
+					},
+				],
+				errorElement: <PortalMessagePage title="Fail to load application" type={'Error'} />,
+			},
+			{
+				path: `/aka/*`,
```
Comment (Gustav-Eikaas) [inline client/packages/portal-client/src/components/portal-router/PortalRouter.tsx:45]: Why is aka/* a route?

File: client/packages/portal-client/src/framework-config/portal-framework-config.tsx [hunk 2]
```diff
@@ -45,11 +46,29 @@ export function createPortalFramework(portalConfig: PortalConfig) {
...
+
+			fusion.context.currentContext$.pipe(skip(1)).subscribe((context) => {
+				const { navigator } = fusion.navigation;
+
+				if (!context) {
+					navigator.replace('/');
+				}
+
+				if (context && context.id && !window.location.pathname.includes(context.id)) {
+					const pathname = window.location.pathname.replace(
```
Comment (Gustav-Eikaas) [inline client/packages/portal-client/src/framework-config/portal-framework-config.tsx:60]: Please extract the regex expression to a separate function and give the function a name explaining what the regex does

File: client/packages/portal-core/src/app-loader/AppNotAwaitable.tsx [hunk 3]
```diff
@@ -1,28 +1,33 @@
...
\ No newline at end of file
+	return (
+		<PortalMessagePage title={`${name} is not awaitable for the selected context`} type="Warning">
+			<div>
+				<ContextProvider>
+					<ContextSelector navigate={navigate} path={`/${currentView?.key}`} />
+				</ContextProvider>
+				{!isLoading && data && (
+					<div>
+						<p>Current apps are awaitable for the selected context</p>
```
Comment (Gustav-Eikaas) [inline client/packages/portal-core/src/app-loader/AppNotAwaitable.tsx:24]: typo awaitable, I suppose you mean available

File: client/packages/portal-core/src/app-loader/use-app-loader.ts [hunk 4]
```diff
@@ -21,7 +21,9 @@ export const useAppLoader = (appKey: string) => {
 		subscription.add(
 			currentApp?.initialize().subscribe({
 				next: ({ manifest, script, config }) => {
-					const basename = window.location.pathname;
+					/** generate basename for application */
+					const [basename] = window.location.pathname.match(/\/?apps\/[a-z|-]+\//g) ?? [''];
```
Comment (Gustav-Eikaas) [inline client/packages/portal-core/src/app-loader/use-app-loader.ts:25]: Please describe all regex

File: client/packages/portal-core/src/utils/utils.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+	return `${path}/${contextId}`;
+}
+
+export function getContextPageUrl(contextId?: string): string {
+	if (!contextId) return 'context-page/';
+	return `/context-page/${contextId}`;
+}
+
+export function getContextFormUrl() {
+	const match = window.location.pathname.match(
```
Comment (Gustav-Eikaas) [inline client/packages/portal-core/src/utils/utils.ts:12]: Is regex really needed to get the context from url?

**Comment by Gustav-Eikaas** [comment]
Two notes/questions

- Im seeing a lot of regex, whats up with that? Is there a better way to do this?
- Why are we not using some form of router in the portal, im seeing a lot of window.location

File: client/packages/portal-core/src/utils/utils.ts (hunk 5 above)
Comment (Noggling) [reply client/packages/portal-core/src/utils/utils.ts:12]: This i s used outside on the routing so its ider this of sting splitting. and if checks

File: client/packages/portal-client/src/components/portal-router/PortalRouter.tsx (hunk 1 above)
Comment (Noggling) [reply client/packages/portal-client/src/components/portal-router/PortalRouter.tsx:45]: Im not sure if this is the right one  but there will be a rout used for applications to link to other application. Fusion Clasic has an aka route, but im not use if it's aka of go_to we are going to use.

**Comment by Noggling** [comment]
> * Im seeing a lot of regex, whats up with that? Is there a better way to do this?
> * Why are we not using some form of router in the portal, im seeing a lot of window. Location

Sorry to say but i dont think there is any better way. all code using regex is outside of react scope so routing is not an option

File: client/packages/portal-pages/src/pages/project-page/ProjectPage.tsx [hunk 6]
```diff
@@ -27,10 +28,21 @@ type ProjectMaster = {
...
-export const ContextPage = () => {
+export const ProjectPage = () => {
+	const { contextId } = useParams();
+
 	const currentContext = useFrameworkCurrentContext<ProjectMaster>();
 
-	if (!currentContext) return null;
+	if (
+		!currentContext ||
+		!contextId?.match(/[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}/)
```
Comment (Gustav-Eikaas) [inline client/packages/portal-pages/src/pages/project-page/ProjectPage.tsx:38]: Extract to a function

if(!parseContext){
return null
}

**Review by Gustav-Eikaas — APPROVED** [review]

**Pull request merged**